src/services/completions.ts

```diff
@@ -167,15 +167,17 @@ namespace ts.Completions {
         const uniqueNames = createMap<true>();
         if (symbols) {
             for (const symbol of symbols) {
-                const entry = createCompletionEntry(symbol, location, performCharacterChecks, typeChecker, target, allowStringLiteral);
-                if (entry) {
-                    const id = entry.name;
-                    if (!uniqueNames.has(id)) {
-                        if (symbolToOriginInfoMap && symbolToOriginInfoMap[getUniqueSymbolId(symbol, typeChecker)]) {
-                            entry.hasAction = true;
+                if (!isEscapedNameOfWellKnownSymbol(symbol.escapedName)) {

```

Comment:
```ts
if (isEscapedNameOfWellKnownSymbol(symbol.escapedName)) {
  continue;
}
```
to avoid the extra nesting? Likewise some of the conditions on the other nested `if` statements could be inverted and used with a `continue` to reduce the nesting a bit here.